components/handlers/module_edit.py

```diff
@@ -8,61 +8,114 @@
-                outcome = model.update_module(module_code, module_name, module_desc, module_mc)
+        outcome = True
+        module_info = model.get_module(module_code)
+        if module_info is None:
+            outcome = False
+        elif (module_name != old_module_name or module_desc != old_module_desc or
+              int(module_mc) != int(old_module_mc)):
+            model.store_original_module_info(module_code, old_module_name,
+                                             old_module_desc, old_module_mc)
+            outcome = model.update_module(module_code, module_name, module_desc, module_mc)
 
-            if not test_data:
-                return Outcome().POST("edit_module", outcome, module_code)
+        if not test_data:
+            return Outcome().POST("edit_module", outcome, module_code)
 
 
 class EditMountingInfo(object):
     '''
-        This class handles the editing of mounting info
+        This class handles the editing of specific mounting info
         (Mounting status and quota)
     '''
+    def __init__(self):
+        '''
+            Define the AY-Sems that are in the system
+            By right, these settings hould be set by the superadmin
+        '''
+        self.number_of_future_ays = 1
+        self.current_ay = model.get_current_ay()
+        self.list_of_future_ay_sems = []
+
+        ay = self.current_ay
+        for i in range(self.number_of_future_ays):
+            ay = model.get_next_ay(ay)
+            self.list_of_future_ay_sems.append(ay+" Sem 1")
+            self.list_of_future_ay_sems.append(ay+" Sem 2")
+
+
+    def GET(self):
+        '''
+            Handles the loading of the 'Edit Specific Module Info' page
+        '''
+        data = web.input()
+        module_code = data.code
+        ay_sem = data.aySem
+
+        module_info = model.get_module(module_code)
+        if module_info is None:
+            return RENDER.notfound("Module " + module_code + " does not exist in the system.")
+        if ay_sem not in self.list_of_future_ay_sems:
+            return RENDER.notfound(ay_sem + " is not in the system's list of future AY-Semesters.")
+
+        module_ay_sem_info_handler = IndividualModule()
+        module_ay_sem_info_handler.load_mounting_info(module_code, ay_sem)
+        mounting_value = module_ay_sem_info_handler.get_mounting_status()
+        quota = module_ay_sem_info_handler.get_quota()
+        overlapping_list = model.get_mod_taken_together_with(module_code)
+
+        return RENDER.mountingEdit(module_code, ay_sem, mounting_value, quota, overlapping_list)
+
+
     def POST(self):
         '''
-            Handles the loading and submission of the 'edit mounting' form
+            Handles the submission of the 'Edit Specific Module Info' page
         '''
         data = web.input()
-        form_status = data.status
         module_code = data.code
         ay_sem = data.aySem
         try:
```

Comment:
Case of exceptional quota values falling outside of permissible range is not handled here.